myapp/app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,37 @@
+class SessionsController < ApplicationController
+  skip_before_action :login_required
+
+  def new
+  end
+
+  def create
+    user = User.find_by(email: session_params[:email])
+
+    if user.present?
+      pass_digest = Digest::MD5.hexdigest(session_params[:password])
+      password = Digest::MD5.hexdigest(pass_digest + user.salt)
+      if user.password == password
+        session[:user_id] = user.id
+        redirect_to root_url, notice: 'ログインしました。'
+      else
+        flash.now[:alert] = 'ログインに失敗しました。'
+        render :new
+      end
+    else
+      flash.now[:alert] = 'ログインに失敗しました。'
+      render :new
+    end
```

Comment:
ネストが気になったので下記のようにした方が順に処理され可読性が上がるかと思うのですがどうでしょうか？
```suggestion
    if user.blank?
      flash.now[:alert] = 'ログインに失敗しました。'
      return render :new
    end

    pass_digest = Digest::MD5.hexdigest(session_params[:password])
    password = Digest::MD5.hexdigest(pass_digest + user.salt)
    if user.password == password
      session[:user_id] = user.id
      redirect_to root_url, notice: 'ログインしました。'
    else
      flash.now[:alert] = 'ログインに失敗しました。'
      render :new
    end
```

Reply:
エラーを回避するため、ネストした状態になっていました。
returnを記載してエラー解消できることを理解し、ネスト構造になっていた部分を修正しました。
（ログイン画面でログインに失敗した際にも各項目が日本語表示されるよう処理自体をrenderではなくredirect_toに変更しました。）